ext/ITensorNetworksEinExprsExt/src/ITensorNetworksEinExprsExt.jl

```diff
@@ -0,0 +1,66 @@
+module ITensorNetworksEinExprsExt
+
+using ITensors
+using ITensorNetworks
+using ITensorNetworks: Index
+using EinExprs: EinExprs, EinExpr, einexpr, SizedEinExpr
+
+function EinExprs.einexpr(tn::ITensorNetwork; optimizer::EinExprs.Optimizer)
+  IndexType = Any
+  T = vertextype(tn)
+
+  tensors = EinExpr{IndexType}[]
+  tensor_map = Dict{Set{IndexType},T}()
+  sizedict = Dict{IndexType,Int}()
+
+  for (key, tensor) in pairs(vertex_data(tn))
+    _inds = collect(inds(tensor))
+    push!(tensors, EinExpr{IndexType}(; head=_inds))
+    tensor_map[Set(_inds)] = key
+    merge!(sizedict, Dict(_inds .=> size(tensor)))
+  end
+
+  _openinds = collect(externalinds(tn))
+  expr = SizedEinExpr(sum(tensors; skip=_openinds), sizedict)
+
+  return einexpr(optimizer, expr)
+end
+
+function ITensorNetworks.contraction_sequence(
+  alg::ITensorNetworks.Algorithm"einexpr", tn::Vector{ITensor}; kwargs...
+)
+  return contraction_sequence(alg, ITensorNetwork(tn); kwargs...)
+end
+
+function ITensorNetworks.contraction_sequence(
+  ::ITensorNetworks.Algorithm"einexpr",
+  tn::ITensorNetwork{T};
+  optimizer=EinExprs.Exhaustive(),
+) where {T}
+  IndexType = Any
+
+  tensors = EinExpr{IndexType}[]
+  tensor_map = Dict{Set{IndexType},T}()
+  sizedict = Dict{IndexType,Int}()
+
+  for (key, tensor) in pairs(vertex_data(tn))
+    _inds = collect(inds(tensor))
+    push!(tensors, EinExpr{IndexType}(; head=_inds))
+    tensor_map[Set(_inds)] = key
+    merge!(sizedict, Dict(_inds .=> size(tensor)))
+  end
+
+  _openinds = collect(externalinds(tn))
+  expr = SizedEinExpr(sum(tensors; skip=_openinds), sizedict)
+
+  path = einexpr(optimizer, expr)
+
+  function _convert_to_contraction_sequence(subpath)
+    EinExprs.nargs(subpath) == 0 && return tensor_map[Set(subpath.head)]
+    return map(_convert_to_contraction_sequence, EinExprs.args(subpath))
+  end
```

Comment:
I'm confused why this needs to be a function.

Reply:
Because it is recursive: it calls itself on line 60.

Reply:
Oh, I see. Maybe we could split this off into a function `to_contraction_sequence(tensor_map, subpath)`? I usually prefer not defining functions within other functions like this since generally they can be factored out into separate functions that may be useful in other places with a bit of thought.

Reply:
Yep, no problem.